nilearn/image/tests/test_image.py

```diff
@@ -654,6 +654,24 @@ def test_new_img_like():
     assert_array_equal(get_data(img_nifti2), get_data(img2_nifti2))
 
 
+def test_new_img_like_accepts_paths(tmp_path):
+    """Check that new_img_like can accept instances of pathlib.Path."""
+    nifti_path = tmp_path / "sample.nii"
+    assert isinstance(nifti_path, Path)
+
+    data = np.random.rand(10, 10, 10)
+    img = Nifti1Image(data, np.eye(4))
```

Comment:
not needed for now but I think there may be a lot random img creation in the whole test suite that could be fixturised

Reply:
Linking #3725 and #3331